recipes/mic.lua

```diff
@@ -0,0 +1,168 @@
+--[[
+
+        Licensed under GNU General Public License v2
+        * (c) 2021, bzgec
+
+
+# Microphone state widget/watcher
+
+This widget can be used to display the current microphone status.
+
+## Requirements
+
+- `amixer` - this command is used to get and toggle microphone state
+
+## Usage
+
+- Download [mic.lua](https://awesomewm.org/recipes/mic.lua) file and put it into awesome's
+  folder (like `~/.config/awesome/widgets/mic.lua`)
+
+- Add widget to `theme.lua`:
+
+```lua
+local widgets = {
+    mic = require("widgets/mic"),
+}
+theme.mic = widgets.mic({
+    timeout = 10,
+    settings = function(self)
+        if self.state == "muted" then
+            self.widget:set_image(theme.widget_micMuted)
+        else
+            self.widget:set_image(theme.widget_micUnmuted)
+        end
+    end
+})
+local widget_mic = wibox.widget { theme.mic.widget, layout = wibox.layout.align.horizontal }
+```
+
+- Create a shortcut to toggle microphone state (add to `rc.lua`):
+
+```lua
+-- Toggle microphone state
+awful.key({ modkey, "Shift" }, "m",
+          function ()
+              beautiful.mic:toggle()
+          end,
+          {description = "Toggle microphone (amixer)", group = "Hotkeys"}
+),
+```
+
+- You can also add a command to mute the microphone state on boot. Add this to your `rc.lua`:
+
+```lua
+-- Mute microphone on boot
+beautiful.mic:mute()
+```
+
+--]]
+
+
+local awful   = require("awful")
+local naughty = require("naughty")
+local gears   = require("gears")
+local wibox   = require("wibox")
+
+local function factory(args)
+    local args = args or {}
+
+    local mic = {
+        widget   = args.widget or wibox.widget.imagebox(),
+        settings = args.settings or function(self) end,
+        timeout  = args.timeout or 10,
+        timer    = gears.timer,
+        state    = "",
+    }
+
+    function mic:mute()
+        awful.spawn.easy_async_with_shell("amixer set Capture nocap",
+            function()
+                self:update()
+            end
+        )
+    end
+
+    function mic:unmute()
+        awful.spawn.easy_async_with_shell("amixer set Capture cap",
+            function()
+                self:update()
+            end
+        )
+    end
+
+    function mic:toggle()
+        awful.spawn.easy_async_with_shell("amixer set Capture toggle",
```

Comment:
All of these do not need a shell, I think.

Reply:
I used a shell in order to wait before calling a `self.update()`. This way `update()` is always called after the command (in this case `amixer set Capture toggle`.

Reply:
What I mean: `easy_async` instead of `easy_async_with_shell`.

The `_with_shell` variant starts `sh -c "your command here"`, which is one extra process to create (first `sh`, then `sh` starts `amixer`).

Reply:
Oh sorry, may bad. You are right. 
It works with `easy_async`. So now it would also be better to use `{"amixer", "set", "Capture", "toggle"}` right?

Reply:
Feel free to split it up like that, but here it does not matter that much. In #156 the invocation was more complicated and you had spaces inside of arguments and needed to escape things. There, I felt like the array version is more readable.

Reply:
Oh I see, but isn't this just a little bit faster (not that it is going to show anywhere, but just saving a bit of processing power)?

Reply:
True :-D